lxd/certificates.go

```diff
@@ -1039,3 +1034,26 @@ func certificateDelete(d *Daemon, r *http.Request) response.Response {
 
 	return response.EmptySyncResponse
 }
+
+func certificateValidate(cert *x509.Certificate) error {
+	if time.Now().Before(cert.NotBefore) {
+		return fmt.Errorf("The provided certificate isn't valid yet")
+	}
+
+	if time.Now().After(cert.NotAfter) {
+		return fmt.Errorf("The provided certificate is expired")
+	}
+
+	if cert.PublicKeyAlgorithm == x509.ECDSA {
+		pubKey, ok := cert.PublicKey.(*rsa.PublicKey)
+		if !ok {
+			return fmt.Errorf("Unable to validate the RSA certificate")
+		}
+
+		if pubKey.Size() < 2048 {
```

Comment:
Looks like an `else if` branch was accidentally omitted.

Reply:
This seems fine as the !ok check will return so won't evaluate here.

Reply:
@tomponline isn't it inside `if cert.PublicKeyAlgorithm == x509.ECDSA {` ?

Reply:
Hrm maybe it should be `cert.PublicKeyAlgorithm != x509.ECDSA`

There's no comments so can't tell what the intention was.

Reply:
It should have been == x509.RSA, fixed now :)